language/stdlib/modules/compare.move

```diff
@@ -0,0 +1,99 @@
+// This defines a total order `Compare::cmp<T>` among Move values of the same type `T`. That is, we
+// have the guarantee:
+// `Compare::cmp<T>(t1, t2) == LESS_THEN` iff `Compare::cmp<T>(t2, t1) == GREATER_THEN`
```

Comment:
THEN -> THAN everywhere

Reply:
I was not able to convince myself that the guarantee is an exact characterization of a total order.  In particular, I don't see how the formal statement of the guarantees ensures that cmp is transitive.